code/modules/client/preferences/preferences.dm

```diff
@@ -0,0 +1,143 @@
+/datum/preferences
+	var/ckey_owner
+	var/max_slots = 3
+	var/list/slots
+
+	var/static/list/selected_slots
+	var/static/selected_slots_dirty = FALSE
+
+/datum/preferences/New(client/owner)
+	ckey_owner = owner.ckey
+	if(IsGuestKey(ckey_owner))
+		max_slots = 1
+	else if(owner.IsByondMember())
+		max_slots = 8
+	GLOB.preferences_datums[ckey_owner] = prefs
+	if(SSdbcore.initialized)
+		Load()
+
+/datum/preferences/Destroy()
+	QDEL_LIST(slots)
+	GLOB.preferences_datums -= ckey_owner]
+	return ..()
+
+/datum/preferences/proc/Load()
+	if(IsGuestKey(ckey_owner))
+		slots = list(new /datum/preference_slot)
+		return
+
+	if(SSdbcore.Connect())
+		var/datum/DBQuery/Q = SSdbcore.NewQuery("SELECT id, json FROM [format_table_name("preferences")] WHERE owner = '[ckey_owner]'")
+		if(Q.Execute() && Q.NextRow())
+			do

```

Comment:
![cmon](http://www.startrek.com/uploads/assets/articles/q-who1.jpg)

Reply:
https://github.com/tgstation/tgstation/blob/a82690cd5e075b2856ff598fe673ca6ba72c95ab/code/controllers/subsystem/dbcore.dm#L78-L79

:?

Reply:
oh whoops

Reply:
Tbh it was more directed at the do-while loop and lame var names.